mirgecom/multiphysics/phenolics.py

```diff
@@ -0,0 +1,84 @@
+""":mod:`mirgecom.multiphysics.phenolics` handles phenolics modeling."""
+
+__copyright__ = """
+Copyright (C) 2023 University of Illinois Board of Trustees
+"""
+
+__license__ = """
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+The above copyright notice and this permission notice shall be included in
+all copies or substantial portions of the Software.
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+THE SOFTWARE.
+"""
+
+from meshmode.dof_array import DOFArray
+from pytools.obj_array import make_obj_array
+from mirgecom.fluid import ConservedVars
+
+
+def initializer(dcoll, gas_model, wall_density, temperature,
+                gas_density=None, pressure=None):
+    """Initialize state of composite material.
+
+    Parameters
+    ----------
+    gas_model
+        :class:`mirgecom.gas_model.GasModel`
+
+    wall_density: numpy.ndarray
+        The initial bulk density of each one of the resin constituents.
+        It has shape ``(nphase,)``
+
+    temperature: :class:`~meshmode.dof_array.DOFArray`
+        The initial temperature of the gas+solid
+
+    gas_density: :class:`~meshmode.dof_array.DOFArray`
+        Optional argument with the gas density. If not provided, the pressure
+        will be used to evaluate the density.
+
+    pressure: :class:`~meshmode.dof_array.DOFArray`
+        Optional argument with the gas pressure. It will be used to evaluate
+        the gas density.
+
+    Returns
+    -------
+    cv: :class:`mirgecom.fluid.ConservedVars`
+        The conserved variables of the fluid permeating the porous wall.
+    """
+    if gas_density is None and pressure is None:
+        raise ValueError("Must specify one of 'gas_density' or 'pressure'")
+
+    if isinstance(temperature, DOFArray) is False:
```

Comment:
```suggestion
    if not isinstance(temperature, DOFArray):
```

Reply:
Done in [bc52690](https://github.com/illinois-ceesd/mirgecom/pull/875/commits/bc52690831096a0ecc59b84d502380dc28fdf6a9)